### PR TITLE
[libevent] Fix 'install failure empty bin/ folder' 

### LINKED ISSUE #20945: [libevent] install failure empty bin/ folder
**Host Environment**
- OS: Linux

**To Reproduce**
It is only reproducible with `x64-linux-dynamic` triplet:
`./vcpkg install libevent:x64-linux-dynamic`

**Failure logs** 
```
-- Performing post-build validation
There should be no empty directories in /home/silverqx/Code/c/vcpkg_test/packages/libevent_x64-linux-dynamic
The following empty directories were found:

/home/silverqx/Code/c/vcpkg_test/packages/libevent_x64-linux-dynamic/bin

If a directory should be populated but is not, this might indicate an error in the portfile.
If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:

file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/a/dir" "${CURRENT_PACKAGES_DIR}/some/other/dir")


Found 1 error(s). Please correct the portfile:
```

**Additional context**
I have temporary added to `libevent/portfile.cmake`:
```
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
```

This problem is specific to Linux only, on Windows and MSYS2 `bin/` folder contains `.dll` libs.
What would be the best solution to run this command on UNIX systems only ? I'm thinking about `if(UNIX)`, `if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")` or do you have any better solution for this?

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fix #20945 

added to file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin") libevent/portfile.cmake to slove
the install failure empty bin/ folder error

All feature are tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

### CODE REVIEW

**Review by silverqx — APPROVED** [review]
LGTM

File: ports/libevent/portfile.cmake [hunk 1]
```diff
@@ -40,17 +39,23 @@ vcpkg_configure_cmake(
...
 
-file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/tools/libevent/)
-file(RENAME ${CURRENT_PACKAGES_DIR}/bin/event_rpcgen.py ${CURRENT_PACKAGES_DIR}/tools/libevent/event_rpcgen.py)
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/libevent/")
+file(RENAME "${CURRENT_PACKAGES_DIR}/bin/event_rpcgen.py" "${CURRENT_PACKAGES_DIR}/tools/libevent/event_rpcgen.py")
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
+if(UNIX)
+    if(CMAKE_HOST_SYSTEM_NAME STREQUAL "Linux")
```
[the remark this answers is not in the document]
Comment (JonLiu1993) [reply ports/libevent/portfile.cmake:50]: @silverqx ,Are you tested it?

**Comment by JonLiu1993** [comment]
@dg0yt @silverqx ,Thanks for your help!

**Comment by silverqx** [comment]
I have increased port-version to 4, should I leave it at 3 and only run `x-add-version`? Or where was the problem? @JonLiu1993

**Comment by silverqx** [comment]
Because when I think about that you have increased port-version from 2 to 3, so from the vcpkg perspective would be enough to only run x-add-version.

**Comment by JonLiu1993** [comment]
> Because when I think about that you have increased port-version from 2 to 3, so from the vcpkg perspective would be enough to only run x-add-version.

Thanks for your reminder, I have corrected it

**Comment by silverqx** [comment]
Heptic monday 😂

**Comment by silverqx** [comment]
It fails at some evpp and folly packages :/

**Comment by JonLiu1993** [comment]
> It fails at some evpp and folly packages :/

I will deal with this error after I finish urgent matters

**Comment by silverqx** [comment]
Problem is that this is missing in vcpkg.json:
```
+  "dependencies": [
+    {
+      "name": "vcpkg-cmake",
+      "host": true
+    },
+    {
+      "name": "vcpkg-cmake-config",
+      "host": true
+    }
+  ],
```

**Comment by JonLiu1993** [comment]
> Problem is that this is missing in vcpkg.json:
> 
> ```
> +  "dependencies": [
> +    {
> +      "name": "vcpkg-cmake",
> +      "host": true
> +    },
> +    {
> +      "name": "vcpkg-cmake-config",
> +      "host": true
> +    }
> +  ],
> ```

@silverqx ,I have already added these two dependent ports in vcpkg.json. I see that the log is related to its other two ports folly and evpp.

**Comment by silverqx** [comment]
@JonLiu1993 you are right, I didn't notice that dependencies are in vcpkg.json, my fault, so I don't know why it is crashing, maybe some problems in folly and evpp, as you wrote you will look later.

**Comment by dg0yt** [comment]
It is a prefix issue in the exported config again:

~~~
CMake Error at /mnt/vcpkg-ci/installed/x64-linux/share/libevent/LibeventTargets-static.cmake:97 (message):
  The imported target "libevent::core" references the file
     "/debug/lib/libevent_core.a"
  but this file does not exist.  Possible reasons include:
  * The file was deleted, renamed, or moved to another location.
  * An install or uninstall procedure did not complete successfully.
  * The installation package was faulty and contained
     "/mnt/vcpkg-ci/installed/x64-linux/share/libevent/LibeventTargets-static.cmake"
  but not all the files it references.
~~~

**Comment by silverqx** [comment]
This is how the import prefix looks like:
```
# The installation prefix configured by this project.
set(_IMPORT_PREFIX "${_IMPORT_PREFIX}")
```

And that is all, I'm pretty sure that bad, it should be something like this:
```
# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()
```

**Comment by silverqx** [comment]
It is of course related to the `vcpkg_cmake_config_fixup` (changed from vcpkg_fixup_cmake_targets).

**Comment by silverqx** [comment]
Ok, now that import prefix looks correct:
```
# Compute the installation prefix relative to this file.
get_filename_component(_IMPORT_PREFIX "${CMAKE_CURRENT_LIST_FILE}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
get_filename_component(_IMPORT_PREFIX "${_IMPORT_PREFIX}" PATH)
if(_IMPORT_PREFIX STREQUAL "/")
  set(_IMPORT_PREFIX "")
endif()
```

Looks promising, will see.

**Comment by JonLiu1993** [comment]
@dg0yt @silverqx ,Thanks agin.

**Review by silverqx — APPROVED** [review]
Everything looks correct to me, I also have tried to build `getdns` port which depends on libevent and it was without problems, ci/cd did this with evpp and folly packages. It was not that easy as it looked like 🙂, thx @JonLiu1993 @dg0yt

**Comment by JonLiu1993** [comment]
@silverqx ,I have a question, does libevent upstream support dynaimic compilation under Linux?

**Comment by silverqx** [comment]
I'm pretty sure that yes because mysqlclient library depends on libevent and mysqlclient library supports dynamic compilation.

**Comment by silverqx** [comment]
When I look at my gentoo's `/usr/lib64/` there are libevent*.so libraries.

**Comment by JonLiu1993** [comment]
> I'm pretty sure that yes because mysqlclient library depends on libevent and mysqlclient library supports dynamic compilation.

Thanks for your clarification

File: ports/libevent/portfile.cmake [hunk 2]
```diff
@@ -40,17 +40,17 @@ vcpkg_configure_cmake(
...
+file(MAKE_DIRECTORY "${CURRENT_PACKAGES_DIR}/tools/libevent/")
+file(RENAME "${CURRENT_PACKAGES_DIR}/bin/event_rpcgen.py" "${CURRENT_PACKAGES_DIR}/tools/libevent/event_rpcgen.py")
 
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
-file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/share)
-if(VCPKG_LIBRARY_LINKAGE STREQUAL static)
-    file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/bin ${CURRENT_PACKAGES_DIR}/debug/bin)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+if(NOT VCPKG_TARGET_IS_WINDOWS OR VCPKG_LIBRARY_LINKAGE STREQUAL static)
```
Comment (dg0yt) [inline ports/libevent/portfile.cmake:52]: `"static"`

File: ports/libevent/portfile.cmake [hunk 3]
```diff
@@ -63,10 +63,8 @@ vcpkg_replace_string(${CURRENT_PACKAGES_DIR}/share/libevent/LibeventTargets-${_t
...
 )
-
+vcpkg_fixup_pkgconfig()
 vcpkg_copy_pdbs()
 
 #Handle copyright
-file(INSTALL ${SOURCE_PATH}/LICENSE DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)
-
-vcpkg_fixup_pkgconfig()
+file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
```
Comment (dg0yt) [inline ports/libevent/portfile.cmake:70]: Please add an EOL at EOF.

**Review by silverqx — APPROVED** [review]
👍 thx

**Comment by JonLiu1993** [comment]
@NancyLi1013 ,Could you please review this pr? thanks.

**Review by NancyLi1013 — APPROVED** [review]

**Pull request merged**